File: DependencyInjection/ChangingBehaviorThroughComposition/Alarms/SilentAlarm.cs
using System;

namespace ChangingBehaviorThroughComposition
{
  public class SilentAlarm : Alarm
  {
    private readonly string _phoneNumber;

    public SilentAlarm(string phoneNumber)
    {
      _phoneNumber = phoneNumber;
    }

    public void Trigger()
    {
      Console.WriteLine("Calling " + _phoneNumber);
    }

    public void Disable()
    {
      Console.WriteLine("Not calling " + _phoneNumber);
    }
  }
}

File: Java/dependency-injection/_00_why-composition/src/main/java/SilentAlarm.java
public class SilentAlarm
  {
    private final String _phoneNumber;

    public SilentAlarm(String phoneNumber)
    {
      _phoneNumber = phoneNumber;
    }

    public void trigger()
    {
      System.out.println("Calling " + _phoneNumber);
    }

    public void disable()
    {
      System.out.println("Not calling " + _phoneNumber);
    }
  }
